app/models/measure_condition_classification.rb

```diff
@@ -0,0 +1,46 @@
+class MeasureConditionClassification
+  DOCUMENT_CLASS = 'document'.freeze
+  EXEMPTION_CLASS = 'exemption'.freeze
+  THRESHOLD_CLASS = 'threshold'.freeze
+  NEGATIVE_CLASS = 'negative'.freeze
+  UNKNOWN_CLASS = 'unknown'.freeze
+
+  EXCLUDED_DOCUMENT_CODE_CLASS_DOCUMENT_CODES = [
+    '999L', # National Document: CDS universal waiver
+    'C084', # Other certificates: Exemption by virtue of Articles 3 and 4 of regulation 2019/2122 (animals intended for scientific purposes, research and diagnostic samples)
+  ].freeze
+
+  EXEMPTION_CERTIFICATE_TYPE_CODE = 'Y'.freeze # Particular provisions
+
+  def initialize(measure_condition)
+    @measure_condition = measure_condition
+  end
+
+  def measure_condition_class
+    return THRESHOLD_CLASS if threshold_class?
+    return NEGATIVE_CLASS if negative_class?
+    return EXEMPTION_CLASS if exemption_class?
+    return DOCUMENT_CLASS if document_class?
+
+    UNKNOWN_CLASS
+  end
```

Comment:
Just for fun: to avoid this conditional cascade we could do this:

```
def measure_condition_class
  threshold_class || negative_class ||
  exemption_class || document_class ||
  UNKNOWN_CLASS
end

def threshold_class
  @measure_condition.condition_duty_amount.present? && THRESHOLD_CLASS
end

def negative_class
  @measure_condition.measure_action.present? && 
  @measure_condition.measure_action.negative_action? &&
  NEGATIVE_CLASS
end

... 

```

also, I guess this should be related to the ask don't tell principle.

Not very opinionated, up to you.

Reply:
@willfish any opinions? I know you'd written this originally, I just moved its location?

Reply:
So I don't have strong opinions other than I don't see it adding a lot of value in an already complex PR